ndb/src/google/cloud/ndb/model.py

```diff
@@ -439,6 +451,204 @@ def _verify_validator(validator):
 
         return validator
 
+    def __repr__(self):
+        """Return a compact unambiguous string representation of a property.
+
+        This cycles through all stored attributes and displays the ones that
+        differ from the default values.
+        """
+        args = []
```

Comment:
I am not a big fan of this implementation and I think it can be ditched in favor of using `inspect.signature`. This will slow things down, but I'd imagine there are approximately 0 users depending on `repr(MyModel.foo)` (i.e. of a property) to be fast.

Reply:
+1. It seems safe to say that the set of users depending on `repr` for anything other than casual use at the REPL prompt is empty, and performance at the prompt can't matter unless it is absurdly bad.

Reply:
@tseaver PTAL, I switched to the `inspect.signature()` approach and didn't really need to modify the unit tests (other than the `foo` -> `_foo` mangling)